tests/unit/h_vialib/secure/url_test.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime, timedelta
+
+import pytest
+from h_matchers import Any
+
+from h_vialib.exceptions import InvalidToken, MissingToken
+from h_vialib.secure import SecureToken
+from h_vialib.secure.url import SecureURL, ViaSecureURL
+
+
+class TestSecureURL:
+    def test_round_tripping(self, secure_url):
+        url = "http://example.com?a=1&tok.sec=OLD_TOKEN&a=2"
+
+        signed_url = secure_url.create(url, {"extra": "value"}, max_age=10)
+        decoded = secure_url.verify(signed_url)
+
+        assert signed_url == Any.url.matching(url).with_query(
+            [("a", "1"), ("a", "2"), ("tok.sec", Any.string())]
+        )
+        assert decoded == {
+            "url": "http://example.com?a=1&a=2",
+            "extra": "value",
+            "exp": Any.int(),
+        }
```

Comment:
I've gone with simpler tests this time. This one right here really proves just about everything about the happy path. We start off with all the data, create a string which encodes it all and is still a URL, and then we can retrieve it all back out